core/src/view/view.cpp

```diff
@@ -228,18 +233,16 @@ LngLat View::getCenterCoordinates() const {
     return center;
 }
 
-void View::update(bool _constrainToWorldBounds) {
+void View::update() {
 
     m_changed = false;
 
-    if (_constrainToWorldBounds) {
+    if (m_constrainToWorldBounds && m_dirtyConstraint) {
         // Approximate the view diameter in pixels by taking the maximum dimension.
         double viewDiameterPixels = std::fmax(getWidth(), getHeight()) / pixelScale();
         // Approximate the minimum zoom that keeps with view span within the drawable projection area. [1]
-        double minZoom = std::log(viewDiameterPixels / MapProjection::tileSize() + 2) / std::log(2);
-        if (m_zoom < minZoom) {
-            m_zoom = static_cast<float>(minZoom);
-        }
+        m_constrainZoom = std::log(viewDiameterPixels / MapProjection::tileSize() + 1.5f) / std::log(2);
```

Comment:
The constant `2` here is changed without explanation, and the new value allows for some (albeit rare) cases where the map shows blank areas.

Reply:
Ohh I was playing with the values to understand the equation. Will update.

Reply:
And accidentally pushed this.